src/ADAL.PCL/AcquireTokenHandlerBase.cs

```diff
@@ -105,9 +105,15 @@ internal abstract class AcquireTokenHandlerBase
         protected UserIdentifierType UserIdentifierType { get; set; }
 
         protected bool LoadFromCache { get; set; }
-        
+
         protected bool StoreToCache { get; set; }
 
+        private const int ExpirationMarginInMinutes = 5;
+
+        private bool _tokenNearExpiry;
+
+        private bool _tokenExtendedLifeTimeExpired;
```

Comment:
why does acquire token handler care about extended life time? all the logic should be hidden inside token cache.